File: guides/micronaut-eclipsestore-persitence/java/src/test/java/example/micronaut/BaseTest.java
package example.micronaut;

import io.micronaut.test.support.TestPropertyProvider;
import org.junit.jupiter.api.io.TempDir;

import jakarta.validation.constraints.NotNull;
import java.io.File;
import java.util.Collections;
import java.util.Map;
import io.micronaut.core.annotation.NonNull;

abstract class BaseTest implements TestPropertyProvider {

    @TempDir
    static File tempDir;

    @Override
    @NonNull
    public Map<String, String> getProperties() {
        return Collections.singletonMap(
                "microstream.storage.main.storage-directory", tempDir.getAbsolutePath()
        );
    }
}

File: guides/micronaut-microstream-persistence/kotlin/src/test/kotlin/example/micronaut/BaseTest.kt
package example.micronaut

import io.micronaut.test.support.TestPropertyProvider
import org.junit.jupiter.api.io.TempDir
import java.io.File

abstract class BaseTest: TestPropertyProvider {

    companion object {
        @TempDir
        @JvmField
        var tempDir: File? = null
    }

    override fun getProperties(): MutableMap<String, String> {
        return mutableMapOf(
            "microstream.storage.main.storage-directory" to tempDir!!.absolutePath
        )
    }
}